commands/delegate_submission.go

```diff
@@ -0,0 +1,49 @@
+package commands
+
+import (
+	commandspb "code.vegaprotocol.io/go-wallet/internal/proto/commands/v1"
+)
+
+func CheckDelegateSubmission(cmd *commandspb.DelegateSubmission) error {
+	return checkDelegateSubmission(cmd).ErrorOrNil()
+}
+
+func checkDelegateSubmission(cmd *commandspb.DelegateSubmission) Errors {
+	errs := NewErrors()
+
+	if cmd == nil {
+		return errs.FinalAddForProperty("delegate_submission", ErrIsRequired)
+	}
+
+	if cmd.Amount <= 0 {
+		errs.AddForProperty("delegate_submission.amount", ErrIsRequired)
```

Comment:
If the user specifies a negative amount, the error message is going to be `amount is required`, which doesn't aligned. You can use `ErrMustBePositive` instead.

Reply:
Does positive imply non-zero? Because in this case I'd expect to be a non zero positive value.

Reply:
It's a uint64 so it can't be negative. The check it really just for zero. Would you be happier if I made it `a==0`?

Reply:
@jeremyletang Yes positive implies non-zero, we have `ErrMustBePositiveOrZero` if 0 is allowed. 

@peterbarrow yes `a == 0` is okay.